src/lib/data.ts

```diff
@@ -43,9 +46,39 @@ export function setWithoutEmit(key: string, value: any | undefined): any {
     return value;
 }
 
-export function set(key: string, value: any | undefined): any {
+export function set(key: string, value: any | undefined, persist: boolean = false): any {
     setWithoutEmit(key, value);
     event_emitter.emit(key, value);
+    if (persist) {
+        if (!persisting) {
+            // we're not already underway persisting something, so we have to first read the remote values, then update and write back the new...
+            // ... taking into account that we might get asked to persist more values while this is happening...
+            persisting = true;
+            to_be_persisted[key] = value;
+            remote_storage.get("persisted-local-storage").then(
```

Comment:
Instead of getting every time we set, maybe we can just do a `get` when we first load or need to sync? This is presuming we add a mechanism to be notified when we need to update, which I think we can do just by using our existing inter-tab communication system, https://github.com/online-go/online-go.com/blob/devel/src/lib/ITC.ts

Reply:
I had steered away from needing to be notified, hoping that the elegantly-simple remote-storage API would be good enough.

Do you have an API addition in mind, to support notification of remote-storage data changes?

If there was such a thing, I'd be inclined to flatten the persisted-remotely localstorage data, so we get updated when individual elements change, and never have to pass the whole set around.